test/e2e/mapml-viewer/customTCRS.html

```diff
@@ -0,0 +1,271 @@
+<!doctype html>
+<html lang="en">
+
+<head>
+  <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width,initial-scale=1">
+  <title>custom TCRS</title>
+  <script type="module" src="mapml-viewer.js"></script>
+  <script>
+    function build() {
+      let template = {
+        projection: "basic",
+        proj4string: "+proj=utm +zone=33 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs",
+        code: "EPSG:3006",
+        definition: {
+          resolutions: [
+            8192, 4096, 2048, 1024, 512, 256, 128
+          ],
+          origin: [0, 0]
+        }
+      };
+      let cProjection = M.defineCustomProjection(template);
+      let map = document.querySelectorAll("mapml-viewer")[0];
+      map.projection = cProjection;
+
+      let customCBM = {
+        projection: "other",
+        proj4string: "+proj=lcc +lat_1=49 +lat_2=77 +lat_0=49 +lon_0=-95 +x_0=0 +y_0=0 +ellps=GRS80 +datum=NAD83 +units=m +no_defs",
+        code: "EPSG:3978",
+        definition: {
+          origin: [-34655800, 39310000],
+          bounds: L.bounds([[-34655800, -39000000], [10000000, 39310000]]),
+          resolutions: [
+            38364.660062653464,
+            22489.62831258996,
+            13229.193125052918,
+            7937.5158750317505,
+            4630.2175937685215,
+            2645.8386250105837,
+            1587.5031750063501,
+            926.0435187537042,
+            529.1677250021168,
+            317.50063500127004,
+            185.20870375074085,
+            111.12522225044451,
+            66.1459656252646,
+            38.36466006265346,
+            22.48962831258996,
+            13.229193125052918,
+            7.9375158750317505,
+            4.6302175937685215,
+            2.6458386250105836,
+            1.5875031750063502,
+            0.92604351875370428,
+            0.52916772500211673,
+            0.31750063500127002,
+            0.18520870375074083,
+            0.11112522225044451,
+            0.066145965625264591
+          ],
+          crs: {
+            tcrs: {
+              horizontal: {
+                name: "x",
+                min: 0,
+                max: zoom => (M.other.options.bounds.getSize().x / M.other.options.resolutions[zoom]).toFixed()
```

Comment:
This is  not JSON, it's JavaScript; so I was wondering if there's a way we can simplify the format of what we're defining as a parameter i.e. defineCustomProjection(jsonTCRSParametersObject).  Let's discuss  and have a think about this.

Reply:
If a TCRS is defined using JSON there would be certain limits, you can parse most of the data but something like `zoom => (...)` would be impossible to do simply (for security reasons it's not recommended either). Using JS/JS object would let the user utilize the full range of potential in my opinion.

Reply:
You might be able to define something like `zoom => (...)` as an array though but I'm not sure if that makes the format more simple or more complex/tedious.